File: scripts/run_migrations.py
#!/usr/bin/env python3
import os
import psycopg2
from psycopg2 import sql
import sys

# Obtener variable de conexión (Supabase proporciona POSTGRES_URL)
postgres_url = os.getenv('POSTGRES_URL')
if not postgres_url:
    print('[v0] Error: POSTGRES_URL environment variable not set')
    sys.exit(1)

try:
    print('[v0] Connecting to Supabase PostgreSQL...')
    conn = psycopg2.connect(postgres_url)
    cur = conn.cursor()
    print('[v0] Connected successfully!')

    # Ejecutar schema.sql
    print('[v0] Executing schema.sql...')
    with open('scripts/01_schema.sql', 'r') as f:
        schema_sql = f.read()
    
    cur.execute(schema_sql)
    conn.commit()
    print('[v0] Schema created successfully!')

    # Ejecutar seed.sql
    print('[v0] Executing seed.sql...')
    with open('scripts/02_seed.sql', 'r') as f:
        seed_sql = f.read()
    
    cur.execute(seed_sql)
    conn.commit()
    print('[v0] Seed data inserted successfully!')

    cur.close()
    conn.close()
    print('[v0] Database setup complete!')

except Exception as e:
    print(f'[v0] Error: {e}')
    sys.exit(1)

File: scripts/run-migrations.js
import dotenv from 'dotenv';

dotenv.config({ path: '.env.local' });

import { createClient } from '@supabase/supabase-js';
import * as fs from 'fs';
import * as path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseServiceKey) {
  console.error('[v0] Missing Supabase environment variables');
  console.error('[v0] NEXT_PUBLIC_SUPABASE_URL:', supabaseUrl);
  console.error('[v0] SUPABASE_SERVICE_ROLE_KEY:', supabaseServiceKey ? 'SET' : 'NOT SET');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseServiceKey, {
  auth: {
    autoRefreshToken: false,
    persistSession: false,
  },
});

async function runSQL(sql) {
  try {
    const { data, error } = await supabase.rpc('exec_sql', { sql });
    if (error) throw error;
    return data;
  } catch (err) {
    // Intentar ejecutar usando admin API directamente
    const response = await fetch(`${supabaseUrl}/rest/v1/rpc/exec_sql`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${supabaseServiceKey}`,
        'apikey': supabaseServiceKey,
      },
      body: JSON.stringify({ sql }),
    });

    if (!response.ok) {
      console.error('[v0] SQL Error Response:', response.status, response.statusText);
      const errorText = await response.text();
      console.error('[v0] Error Details:', errorText);
      throw new Error(`SQL execution failed: ${response.statusText}`);
    }

    return await response.json();
  }
}

async function executeSQLFile(filePath, description) {
  try {
    console.log(`\n[v0] ${description}...`);
    const fullPath = path.join(__dirname, filePath);
    const sqlContent = fs.readFileSync(fullPath, 'utf-8');

    // Dividir por puntos y coma, eliminando comentarios
    const statements = sqlContent
      .split(';')
      .map(stmt => stmt.trim())
      .filter(stmt => stmt && !stmt.startsWith('--'));

    console.log(`[v0] Found ${statements.length} SQL statements`);

    for (let i = 0; i < statements.length; i++) {
      const stmt = statements[i];
      try {
        console.log(`[v0] Executing statement ${i + 1}/${statements.length}...`);
        await runSQL(stmt);
      } catch (error) {
        console.warn(`[v0] Warning on statement ${i + 1}: ${error.message}`);
        // Continuar con el siguiente statement
      }
    }

    console.log(`[v0] ✓ ${description} completed`);
    return true;
  } catch (error) {
    console.error(`[v0] ✗ Error in ${description}:`, error.message);
    return false;
  }
}

async function main() {
  try {
    console.log('[v0] Starting database migrations...');
    console.log('[v0] Supabase URL:', supabaseUrl);

    const schemaSuccess = await executeSQLFile('01_schema.sql', 'Creating schema');
    const seedSuccess = await executeSQLFile('02_seed.sql', 'Inserting seed data');

    if (schemaSuccess && seedSuccess) {
      console.log('\n[v0] ✓ Database setup complete!');
      process.exit(0);
    } else {
      console.log('\n[v0] ✗ Database setup completed with some warnings');
      process.exit(0); // Salir con éxito igualmente
    }
  } catch (error) {
    console.error('[v0] Fatal error:', error);
    process.exit(1);
  }
}

main();
